apps/emqx/src/emqx_persistent_session_ds_state.erl

```diff
@@ -53,7 +59,7 @@
 
 -type subscriptions() :: emqx_topic_gbt:t(_SubId, emqx_persistent_session_ds:subscription()).
 
--opaque session_iterator() :: emqx_persistent_session_ds:id() | '$end_of_table'.
+-type session_iterator() :: emqx_persistent_session_ds:id() | '$end_of_table'.
```

Comment:
What triggered the change from opaque to transparent? Does the new code rely on the internal structure of the iterator? Ideally it should be avoided, because I am planning to change the storage from mnesia to DS in near future.

Reply:
What triggered this was the check for no persistent sessions, and also if the iteration has reached the end of the table.

Reply:
P.S. I think the return type of `session_interator_next` should be adjusted instead like this:

```
-spec session_iterator_next(session_iterator(), pos_integer()) ->
    {[{emqx_persistent_session_ds:id(), metadata()}], session_iterator() | '$end_of_table'}.
```

Reply:
Good idea, I'll adjust it.

Reply:
This check is just comparing the first batch with `[]` and batch size = 1? I think this can be done in the msgmt code to keep the interface for the CRUD module clean.